src/zjs_fs.c

```diff
@@ -0,0 +1,940 @@
+                                      const jerry_value_t this,
+                                      const jerry_value_t argv[],
+                                      const jerry_length_t argc) {
+    return zjs_readdir(function_obj, this, argv, argc, 0);
+}
+
+static jerry_value_t zjs_readdir_async(const jerry_value_t function_obj,
+                                       const jerry_value_t this,
+                                       const jerry_value_t argv[],
+                                       const jerry_length_t argc) {
+    return zjs_readdir(function_obj, this, argv, argc, 1);
+}
+
+static jerry_value_t zjs_stat(const jerry_value_t function_obj,
+                              const jerry_value_t this,
+                              const jerry_value_t argv[],
+                              const jerry_length_t argc,
+                              uint8_t async)
+{
+    if (!jerry_value_is_string(argv[0])) {
+        ERR_PRINT("first parameter was not path\n");
+        return ZJS_UNDEFINED;
+    }
+    if (async) {
+        if (!jerry_value_is_function(argv[1])) {
+            ERR_PRINT("second parameter was not callback function\n");
+            return ZJS_UNDEFINED;
+        }
+    }
+
+    jerry_size_t jlen = jerry_get_string_size(argv[0]);
+    char path[jlen + 1];
+
+    int wlen = jerry_string_to_char_buffer(argv[0], (jerry_char_t *)path, jlen);
+    if (jlen != wlen) {
+        ERR_PRINT("size mismatch\n");
+        return ZJS_UNDEFINED;
+    }
+    path[wlen] = '\0';
+
+    int ret;
+    struct fs_dirent entry;
+
+    ret = fs_stat(path, &entry);
+
+    if (async) {
+        jerry_value_t args[2];
+
+        args[0] = jerry_create_number(ret);
+        args[1] = create_stats_obj(&entry);
+
+        zjs_callback_id id = zjs_add_callback_once(argv[1],
+                                                   this,
+                                                   NULL,
+                                                   NULL);
+
+        zjs_signal_callback(id, args, sizeof(jerry_value_t) * 2);
+
+        return ZJS_UNDEFINED;
+    } else {
+        return create_stats_obj(&entry);
+    }
+}
+
+static jerry_value_t zjs_stat_sync(const jerry_value_t function_obj,
+                                   const jerry_value_t this,
+                                   const jerry_value_t argv[],
+                                   const jerry_length_t argc) {
+    return zjs_stat(function_obj, this, argv, argc, 0);
+}
+
+static jerry_value_t zjs_stat_async(const jerry_value_t function_obj,
+                                    const jerry_value_t this,
+                                    const jerry_value_t argv[],
+                                    const jerry_length_t argc) {
+    return zjs_stat(function_obj, this, argv, argc, 1);
+}
+
+jerry_value_t zjs_fs_init()
+{
+    jerry_value_t fs = jerry_create_object();
+
+    zjs_obj_add_function(fs, zjs_open_async, "open");
```

Comment:
Follow this pattern to make the changes Jimmy mentions:
`git show d4e1d64 -- 'src/zjs_aio.[ch]'`